java/vector/src/main/java/org/apache/arrow/vector/dictionary/DictionaryEncoder.java

```diff
@@ -114,11 +118,10 @@ public static ValueVector decode(ValueVector indices, Dictionary dictionary) {
     return decoded;
   }
 
-  private static void validateType(MinorType type) {
-    // byte arrays don't work as keys in our dictionary map - we could wrap them with something to
-    // implement equals and hashcode if we want that functionality
-    if (type == MinorType.VARBINARY || type == MinorType.FIXEDSIZEBINARY || type == MinorType.UNION) {
```

Comment:
it seems like there is a regression here for union?

Reply:
Since List, Struct and Binary are supported, I think UNION will work fine, also should add a test. But since we will refactor the encoder with hashtable and hashCode & equals API, I think do this after https://github.com/apache/arrow/pull/4846 maybe better.